app/src/main/java/com/keylesspalace/tusky/components/systemnotifications/NotificationFetcher.kt

```diff
@@ -72,69 +71,41 @@ class NotificationFetcher @Inject constructor(
                     //   (and should therefore adhere to the notification config).
                     eventHub.dispatch(NewNotificationsEvent(account.accountId, notifications))
 
-                    // There's a maximum limit on the number of notifications an Android app
-                    // can display. If the total number of notifications (current notifications,
-                    // plus new ones) exceeds this then some newer notifications will be dropped.
-                    //
-                    // Err on the side of removing *older* notifications to make room for newer
-                    // notifications.
-                    val currentAndroidNotifications = notificationManager.activeNotifications
-                        .sortedWith(
-                            compareBy({ it.tag.length }, { it.tag })
-                        ) // oldest notifications first
-
-                    // Check to see if any notifications need to be removed
-                    val toRemove = currentAndroidNotifications.size + notifications.size - MAX_NOTIFICATIONS
-                    if (toRemove > 0) {
-                        // Prefer to cancel old notifications first
-                        currentAndroidNotifications.subList(
-                            0,
-                            min(toRemove, currentAndroidNotifications.size)
+                    val notificationsByType: Map<Notification.Type, List<Notification>> = notifications.groupBy { it.type }
+
+                    var typeIndex = 0
+                    notificationsByType.forEach { notificationsGroup: Map.Entry<Notification.Type, List<Notification>> ->
+                        // NOTE: Enqueue the summary first: Only this way one can avoid running into notification rate limits;
+                        //   which also would prohibit the group summary to be shown and thus proper grouping!
+                        NotificationHelper.showSummaryNotification(
+                            context,
+                            notificationManager,
+                            account,
+                            notificationsGroup.key,
+                            notificationsGroup.value
                         )
-                            .forEach { notificationManager.cancel(it.tag, it.id) }
-
-                        // Still got notifications to remove? Trim the list of new notifications,
-                        // starting with the oldest.
-                        while (notifications.size > MAX_NOTIFICATIONS) {
-                            notifications.removeAt(0)
-                        }
-                    }
 
-                    val notificationsByType = notifications.groupBy { it.type }
-
-                    // Make and send the new notifications
-                    // TODO: Use the batch notification API available in NotificationManagerCompat
```

Comment:
What about this comment? I think it makes sense.
From the docs of the batch notification api: 
> This is the preferred method for posting groups of notifications, to improve sound and animation behavior.

Reply:
Where can I find the batch notification api?

Reply:
[NotificationManagerCompat.from(context).notify(...)](https://developer.android.com/reference/androidx/core/app/NotificationManagerCompat#notify(java.util.List%3Candroidx.core.app.NotificationManagerCompat.NotificationWithIdAndTag%3E))

Reply:
Done.